compiler/rustc_hir_analysis/src/collect/lifetimes.rs

```diff
@@ -1184,68 +1184,14 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                     break None;
                 }
 
-                Scope::Binder { ref lifetimes, scope_type, s, where_bound_origin, .. } => {
```

Comment:
`where_bound_origin` can be removed completely.

Reply:
https://github.com/rust-lang/rust/blob/226b2496fc3ef997fa1cccf84259539203f82375/compiler/rustc_hir_analysis/src/collect/lifetimes.rs#L1297

The removal of `where_bound_origin` will also remove the fix constructed at https://github.com/rust-lang/rust/pull/97683/files, which seems unrelated to the current feature gate -> https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=774bb194ce32f1c247a1cd54f0d1ec3c